app/src/main/java/com/glia/exampleapp/ExampleCustomMessageAdapter.java

```diff
@@ -0,0 +1,87 @@
+package com.glia.exampleapp;
+
+import android.content.Context;
+import android.content.res.ColorStateList;
+import android.graphics.Typeface;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.core.content.ContextCompat;
+import androidx.core.content.res.ResourcesCompat;
+
+import com.glia.androidsdk.chat.ChatMessage;
+import com.glia.widgets.UiTheme;
+import com.glia.widgets.chat.adapter.CustomMessageAdapter;
+import com.glia.widgets.chat.adapter.holder.CustomMessageViewHolder;
+import com.glia.widgets.chat.adapter.holder.WebViewViewHolder;
+
+import org.json.JSONException;
+
+public class ExampleCustomMessageAdapter extends CustomMessageAdapter {
+    private static final int WEB_VIEW_TYPE = 1;
+    private static final int NATIVE_VIEW_TYPE = 2;
+
+    @Override
+    @Nullable
+    public Integer getItemViewType(ChatMessage message) {
+        if (WebViewViewHolder.isWebViewType(message)) {
+            return WEB_VIEW_TYPE;
+        }
+        return NATIVE_VIEW_TYPE;
+    }
+
+    @Override
+    @NonNull
+    public CustomMessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, @NonNull LayoutInflater inflater,
+                                                      @NonNull UiTheme uiTheme, int viewType) {
+        if (viewType == WEB_VIEW_TYPE) {
+            return new WebViewViewHolder(parent);
```

Comment:
I am not sure I understand this could you please help me? The WebViewHolder is the Widget SDK internal class, not the custom class provided by the integrator? Is this the approach to fall back to the Widget SDK implementation?

Reply:
:thinking:

Reply:
WebViewViewHolder is an implementation of the CustomCardViewHolder. It provides by Widgets SDK to make it easier for UI Custom Card implementation.